internal/engine/local/servercontroller.go

```diff
@@ -37,6 +40,9 @@ type ServerController struct {
 	createdTriggerTime map[string]time.Time
 	client             ctrlclient.Client
 
+	mu         sync.Mutex
+	cmdServers map[string]CmdServer
```

Comment:
this seems really bizarre - why does this PR add new state to the ServerController? i tried reading the code a few times and couldn't figure it out, and also couldn't find any comments to explain it.

Reply:
I added some comments.

It might be more consistent to replicate this back to EngineState rather than storing them here, but I couldn't think of a reason it'd be worth the time/code.

Reply:
ah, ok